### PR TITLE
make sure date is fetched from the db if not available

### PR DESCRIPTION
**Pull request by lucatume**
Tickets:
* https://central.tri.be/issues/76730
* https://central.tri.be/issues/76729

### CODE REVIEW

File: src/Tribe/API.php [hunk 1]
```diff
@@ -271,10 +271,14 @@ protected static function prepare_event_date_meta( $event_id, $data ) {
 
 			if ( isset( $data['EventStartDate'] ) ) {
 				$data['EventStartDate'] = Tribe__Date_Utils::datetime_from_format( $datepicker_format, $data['EventStartDate'] );
+			} elseif ( $existing_start_date = get_post_meta( $event_id, '_EventStartDate', true ) ) {
+				$data['EventStartDate'] = $existing_start_date;
 			}
 
 			if ( isset( $data['EventEndDate'] ) ) {
 				$data['EventEndDate'] = Tribe__Date_Utils::datetime_from_format( $datepicker_format, $data['EventEndDate'] );
+			} elseif ( $existing_end_date = get_post_meta( $event_id, '_EventEndDate', true ) ) {
```
Comment (bordoni) [inline src/Tribe/API.php]: Should we use `tribe_get_event_meta`?

File: src/Tribe/API.php [hunk 2]
```diff
@@ -271,10 +271,14 @@ protected static function prepare_event_date_meta( $event_id, $data ) {
 
 			if ( isset( $data['EventStartDate'] ) ) {
 				$data['EventStartDate'] = Tribe__Date_Utils::datetime_from_format( $datepicker_format, $data['EventStartDate'] );
+			} elseif ( $existing_start_date = get_post_meta( $event_id, '_EventStartDate', true ) ) {
```
Comment (bordoni) [inline src/Tribe/API.php]: Should we use `tribe_get_event_meta`?

Comment (lucatume) [inline src/Tribe/API.php]: I think it adds an overhead we do not need in this case (post id helper and such); it's good in "combat zone": by the time we are here we know the `$event_id` is legit

File: src/Tribe/API.php (hunk 1 above)
Comment (lucatume) [inline src/Tribe/API.php]: same as above

**Pull request merged**